android/library/src/main/java/io/logto/android/Logto.kt

```diff
@@ -67,8 +61,82 @@ class Logto(
         } ?: onComplete(Error("You are not signed in!"))
     }
 
-    private fun clearCredential() {
-        credentialStorage?.clearCredential()
-        credentialCache = null
+    fun getAccessToken(
+        block: (error: Error?, accessToken: String?) -> Unit
+    ) {
+        if (credential == null) {
+            block(Error("Not Sign In!"), null)
+            return
+        }
+
+        if (System.currentTimeMillis() < accessTokenExpiresAt) {
+            block(null, credential?.accessToken)
+            return
+        }
+
+        refreshCredential { error, credential ->
+            block(error, credential?.accessToken)
+        }
+    }
+
+    fun refreshCredential(
+        block: (error: Error?, credential: Credential?) -> Unit
+    ) {
+        if (credential == null) {
+            block(Error("Not Sign In!"), null)
+            return
+        }
+
+        val refreshToken = credential?.refreshToken
+        if (refreshToken == null) {
+            block(Error("Not Support Token Refresh!"), null)
+            return
+        }
+
+        MainScope().launch {
+            try {
+                val updatedCredential = logtoService.refreshCredential(
+                    clientId = logtoConfig.clientId,
+                    redirectUri = logtoConfig.redirectUri,
+                    refreshToken = refreshToken
+                )
+                updateCredential(updatedCredential)
+                block(null, updatedCredential)
+            } catch (error: Error) {
+                block(error, null)
+            }
+        }
+    }
+
+    private val logtoService = LogtoService.create(logtoConfig.oidcEndpoint)
+
+    private var credentialStorage: CredentialStorage? = null
+
+    private var credentialCache: Credential? = null
+        set(value) {
+            if (value != null) {
+                field = credentialCache
+                accessTokenExpiresAt = TokenUtil.expiresAt(value)
+            } else {
+                field = null
+                accessTokenExpiresAt = 0L
+            }
+        }
+
+    private var accessTokenExpiresAt: Long = 0L
+
+    private fun updateCredential(credential: Credential?) {
+        credentialCache = credential
+        if (credential == null) {
+            credentialStorage?.clearCredential()
+        } else {
+            credentialStorage?.saveCredential(credential)
```

Comment:
nit: Just wondering is it a valid case for us, that some times we only need to update a single / partial  params ? i.e （only access_token not refresh_token） in your `credential` object? If so feels like we could have a merge logic implemented within this `updateCredential` method.  For those function caller only need to pass in the params that needs to be updated